src/components/inputs/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,215 @@
+import React, { useState } from "react";
+import styled from "@emotion/styled";
+import { css, Theme } from "@emotion/react";
+import CheckIcon from "components/inputs/TextInput/CheckIcon";
+import { TConditionCheck } from "./types/TConditionCheck";
+
+interface Props {
+  name?: string;
+  label?: string;
+  placeholder?: string;
+  conditionList?: string[];
+  conditionCheckList?: TConditionCheck[];
+  multiline?: boolean;
+  onTextChange?: (value: string, isValid: boolean) => void;
+}
+
+const TextInput: React.FC<Props> = ({
+  name,
+  label,
+  placeholder = "",
+  conditionList,
+  conditionCheckList,
+  multiline = false,
+  onTextChange,
+}) => {
+  const [status, setStatus] = useState("default"); // default / success / invalid / focus
+  const [value, setValue] = useState("");
+  const [isBlurred, setIsBlurred] = useState(false);
+
+  const handleFocus = () => {
+    setStatus("focus");
+  };
+
+  const handleBlur = () => {
+    setIsBlurred(true);
+    if (conditionCheckList ? conditionCheckList.every((check) => check.condition(value)) : true) {
+      setStatus("success");
+      if (onTextChange) {
+        onTextChange(value, true); // Notify parent component that input is valid
+      }
+    } else {
+      setStatus("invalid");
+      if (onTextChange) {
+        onTextChange(value, false); // Notify parent component that input is invalid
+      }
+    }
+  };
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    const newValue = event.target.value;
+    setValue(newValue);
+
+    // 최초 Blur 발생 후, 입력된 텍스트의 조건 확인을 수행하여 isValid 상태 업데이트
+    if (isBlurred && newValue != "") {
+      if (
+        conditionCheckList ? conditionCheckList.every((check) => check.condition(newValue)) : true
+      ) {
+        setStatus("focus");
+      } else {
+        setStatus("invalid");
+      }
+    } else {
+      setStatus("focus");
+    }
+
+    if (onTextChange) {
+      onTextChange(newValue, status != "invalid" && status != "default");
+    }
+  };
+
+  return (
+    <EmotionWrapper>
+      {label && <span className="label">{label}</span>}
+      {conditionList && (
+        <div className="spanList">
+          {conditionList.map((condition, index) => (
+            <span className="condition" key={index}>
+              {condition}
+            </span>
+          ))}
+        </div>
+      )}
+      <div className="inputContainer">
+        {multiline ? (
+          <textarea
+            name={name ? name : label}
```

Comment:
오 name prop 추가해주신 점 확인하였습니다. 그런데 `label` 값으로 fallback 할 필요 없이 그냥 `name` 만 써도 될 것 같습니다..!  (name prop 주어지지 않은 경우 그냥 undefined 로 처리) name prop 을 넣지 않았는데 label 키값으로 name 이 생기면 사용하는 쪽에서 헷갈릴 수 있을 것 같습니다. 

```suggestion
            name={name}
```

Reply:
넵 반영하였습니다!